server/storage/schema/schema_test.go

```diff
@@ -187,17 +189,23 @@ func TestMigrate(t *testing.T) {
 			walEntries: []etcdserverpb.InternalRaftRequest{
 				{ClusterVersionSet: &membershippb.ClusterVersionSetRequest{Ver: "3.6.0"}},
 			},
-			expectVersion:  &version.V3_6,
-			expectError:    true,
-			expectErrorMsg: "cannot downgrade storage, WAL contains newer entries",
+			expectedStorageVersion: &version.V3_6,
+			expectError:            true,
+			expectErrorMsg:         "cannot downgrade storage, WAL contains newer entries",
 		},
 		{
-			name:           "Downgrading v3.5 to v3.4 is not supported as schema was introduced in v3.6",
-			version:        version.V3_5,
-			targetVersion:  version.V3_4,
-			expectVersion:  nil,
-			expectError:    true,
-			expectErrorMsg: `cannot create migration plan: version "3.5.0" is not supported`,
+			name:          "Downgrading v3.5 to v3.4 is supported",
+			version:       version.V3_5,
+			targetVersion: version.V3_4,
+			// note: 3.4 doesn't have storageVersion, this field was added in 3.6
```

Comment:
That's right, could we however also test that `DetectSchemaVersion` will properly recognize v3.4?

Reply:
I've added this, but had to add another case to `UnsafeDetectSchemaVersion`, otherwise this won't work for `Upgrading v3.4 to v3.5 should succeed`

Reply:
I'm little worried about implementing `etcdutl migrate` from v3.4 -> v3.5.